src/pager.c

```diff
@@ -831,14 +831,20 @@ checkfile(char *inp, struct permonst *pm, boolean user_typed_name,
         if (alt && (ap = strstri(alt, " (")) != 0 && ap > alt)
             *ap = '\0';
 
+        /* If the object's name matches the player-specified fruitname,
+           then "fruit" is the alternate description. We do this here so that
+           if the fruit name is an extant object, looking at the fruit yields
+           that object's description. */
+        if (!alt && !strncmpi(dbase_str, gp.pl_fruit, PL_FSIZ))
```

Comment:
If you want to match fruitnames you might want to check `fruit_from_name(dbase_str, TRUE, (int *) 0) != (struct fruit *) 0` so it will include fruits that still exist in-game after fruitname has been changed by the player.